file.go

```diff
@@ -464,24 +402,30 @@ func (f *File) UnsetName(delta *[]*Undo) {
 	u.mod = f.mod
 	u.seq = f.seq
 	u.p0 = 0 // unused
-	u.n = len(f.name)
-	u.buf = []rune(f.name)
-	(*delta) = append(*delta, &u)
+	u.n = len(f.details.Name)
+	u.buf = []rune(f.details.Name)
+	*delta = append(*delta, &u)
 }
 
-func NewFile(filename string) *File {
+func NewFile(filename string) *File { //Todo: Make a friend function that takes the backing RuneArray as input for b and rename function MakeObservableTextBuffer
```

Comment:
`TODO...` And the `TODO` is should be on a line by itself. And please line-break the comment in a reasonable fashion. i.e. follow other examples of `TODO` formatting.